smallrye-reactive-messaging-amqp/src/test/java/io/smallrye/reactive/messaging/amqp/AmqpFailureHandlerTest.java

```diff
@@ -131,6 +140,46 @@ public void testRejectStrategy() {
         assertThat(isAmqpConnectorReady(connector)).isTrue();
     }
 
+    @Test
+    public void testModifiedFailedStrategy() {
+        getModifiedFailedConfig();
+        MyReceiverBeanRecovering bean = deployRecovering();
+        AtomicInteger counter = new AtomicInteger();
+        AmqpConnector connector = container.getBeanManager().createInstance().select(AmqpConnector.class,
+                ConnectorLiteral.of(AmqpConnector.CONNECTOR_NAME)).get();
+        await().until(() -> isAmqpConnectorReady(connector));
+        await().until(() -> isAmqpConnectorAlive(connector));
+
+        usage.produceTenIntegers("modified-failed", counter::getAndIncrement);
+
+        await().atMost(2, TimeUnit.MINUTES).until(() -> bean.list().size() >= 10);
```

Comment:
Seems odd to wait for >=10 but then assert exactly 13 values, this will surely be racey? Particularly as the 10th message will satistfy the await, but is also one of the ones to be redelivered.

Reply:
Good point!